book/src/protocol.md

```diff
@@ -0,0 +1,611 @@
+themselves for a single (accepting) transcript.
+
+> **State Restoration Witness Extended Emulation** Let $\ip$ be an interactive
+> argument for relation $\relation$ with $r = r(\sec)$ challenges. We define for
+> all non-uniform algebraic provers $\alg{\prover}$, extractors $\extractor$,
+> and computationally unbounded distinguishers $\distinguisher$ the advantage
+> metric
+$$
+\adv^\srwee_{\ip, \relation}(\alg{\prover}, \distinguisher, \extractor, \sec) = \textnormal{Pr} \left[ \weereal^{\prover,\distinguisher}_{\ip,\relation}(\sec) \right] - \textnormal{Pr} \left[ \weeideal^{\extractor,\prover,\distinguisher}_{\ip,\relation}(\sec) \right]
+$$
+> is defined with the respect to the following games.
+$$
+\begin{array}{ll}
+\begin{array}{ll}
+&\underline{\bold{Game} \, \weereal_{\ip,\relation}^{\alg{\prover},\distinguisher}(\sec):} \\
+&\tr \gets \epsilon \\
+&\pp \gets \ip.\setup(1^\sec) \\
+&(x, \state{\prover}) \gets \alg{\prover}(\pp) \\
+&\textnormal{Run} \, \alg{\prover}^{\oracle_\real}(\state{\prover}) \\
+&b \gets \distinguisher(\tr) \\
+&\textnormal{Return} \, b = 1 \\
+&\underline{\bold{Game} \, \weeideal_{\ip,\relation}^{\extractor,\alg{\prover},\distinguisher}(\sec):} \\
+&\tr \gets \epsilon \\
+&\pp \gets \ip.\setup(1^\sec) \\
+&(x, \state{\prover}) \gets \alg{\prover}(\pp) \\
+&\state{\extractor} \gets (1^\sec, \pp, x) \\
+&\textnormal{Run} \, \alg{\prover}^{\oracle_\ideal}(\state{\prover}) \\
+&w \gets \extractor(\state{\extractor}, \bottom) \\
+&b \gets \distinguisher(\tr) \\
+&\textnormal{Return} \, (b = 1) \\
+&\, \, \land (\textnormal{Acc}(\tr) \implies (x, w) \in \relation) \\
+\end{array} &
+\begin{array}{ll}
+&\underline{\bold{Oracle} \, \oracle_\real(\tau = (a_1, c_1, ..., a_{i - 1}, c_{i - 1}), a_i):} \\
+& \textnormal{If} \, \tau \in \tr \, \textnormal{then} \\
+& \, \, \textnormal{If} \, i \leq r \, \textnormal{then} \\
+& \, \, \, \, c_i \gets \ch_i; \tr \gets \tr || (\tau, a_i, c_i); \textnormal{Return} \, c_i \\
+& \, \, \textnormal{Else if} \, i = r + 1 \, \textnormal{then} \\
+& \, \, \, \, d \gets \ip.\verifier (\pp, x, (\tau, a_i)); \tr \gets (\tau, a_i) \\
+& \, \, \, \, \textnormal{If} \, d = 1 \, \textnormal{then win} \gets \tt{true} \\
+& \, \, \, \, \textnormal{Return} \, d \\
+& \, \, \textnormal{Return} \, \bottom \\
+\\
+\\
+&\underline{\bold{Oracle} \, \oracle_\ideal(\tau, a):} \\
+& \textnormal{If} \, \tau \in \tr \, \textnormal{then} \\
+& \, \, (r, \state{\extractor}) \gets \extractor(\state{\extractor}, \left[(\tau, a)\right]) \\
+& \, \, \tr \gets \tr || (\tau, a, r) \\
+& \, \, \textnormal{Return} \, r \\
+&\textnormal{Return} \, \bottom
+\end{array}
+\end{array}
+$$
+
+#### Zero Knowledge
+
+We say that an argument of knowledge is _zero knowledge_ if the verifier also
+does not learn anything from their interaction besides that which can be learned
+from the existence of a valid $w$. More formally,
+
+> **Perfect Special Honest-Verifier Zero Knowledge.** A public coin interactive
```

Comment:
```suggestion
> **Perfect Special Honest-Verifier Zero Knowledge.** A public-coin interactive
```

Reply:
Btw, I was for a long time confused about what "special" meant. The answer, as far as I can tell, is that it doesn't mean anything; there is no HVZK property without the "special" modifier. It's just a historical quirk probably coming from the fact that it's often used in conjunction with "special soundness", where the "special" *does* mean something.